### PR TITLE
Updated Flux API

### PR DESCRIPTION
**Pull request by dhairyagandhi96**

### CODE REVIEW

File: other/fizzbuzz/fizzbuzz.jl [hunk 1]
```diff
@@ -49,6 +50,6 @@ end
 
 # Training
 for e in 0:1000
-    train!(loss, [(X, y)], opt)
+    train!(loss, ps, [(X, y)], opt)
```
Comment (MikeInnes) [inline other/fizzbuzz/fizzbuzz.jl]: In some of these simple cases I think we should just inline `ps`.

```suggestion
    train!(loss, params(m), [(X, y)], opt)
```

File: text/treebank/recursive.jl [hunk 2]
```diff
@@ -29,8 +29,10 @@ end
...
-opt = ADAM(params(embedding, W, sentiment))
+opt = ADAM()
+ps = params(embedding, W, sentiment)
 evalcb = () -> @show loss(train[1])
 
-Flux.train!(loss, zip(train), opt,
-            cb = throttle(evalcb, 10))
+Flux.train!(loss, ps, zip(train), opt,
+           cb = throttle(evalcb, 10))
+end
```
Comment (MikeInnes) [inline text/treebank/recursive.jl]: What's this `end` for?

File: vision/mnist/vae.jl [hunk 3]
```diff
@@ -47,19 +47,20 @@ modelsample() = rand.(Bernoulli.(f(z.(zeros(Dz), zeros(Dz)))))
...
   @info "Epoch $i"
-  Flux.train!(loss, zip(data), opt, cb=evalcb)
+  Flux.train!(loss, ps, zip(data), opt, cb=evalcb)
 end
 
 
 ################################# Sample Output ##############################
 
-using Images
+#using Images
```
Comment (MikeInnes) [inline vision/mnist/vae.jl]: Why is this getting commented out?

File: text/treebank/recursive.jl (hunk 2 above)
Comment (dhairyagandhi96) [reply text/treebank/recursive.jl]: Thanks

File: other/fizzbuzz/fizzbuzz.jl (hunk 1 above)
Comment (dhairyagandhi96) [reply other/fizzbuzz/fizzbuzz.jl]: Got it

File: vision/mnist/vae.jl (hunk 3 above)
Comment (dhairyagandhi96) [reply vision/mnist/vae.jl]: This was an artefact of debugging the notebooks generation. Thanks for catching it

**Comment by KristofferC** [comment]
The CUDA versions some of the models (e.g. MNIST conv) doesn't work anymore due to the project files not being updated.

**Comment by MikeInnes** [comment]
Should've caught that. @dhairyagandhi96 can you fix that ASAP?

**Comment by KristofferC** [comment]
Actually, even after updating the packages it doesn't work

```
ERROR: LoadError: GPU compilation of #23(CuArrays.CuKernelState, CUDAnative.CuDeviceArray{Bool,1,CUDAnative.AS.Global}, Base.Broadcast.Broadcasted{Nothing,Tuple{Base.OneTo{Int64}},typeof(==),Tuple{Base.Broadcast.Extruded{CUDAnative.CuDeviceArray{Int64,1,CUDAnative.AS.Global},Tuple{Bool},Tuple{Int64}},Base.Broadcast.Extruded{Array{Int64,1},Tuple{Bool},Tuple{Int64}}}}) failed
KernelError: passing and using non-bitstype argument

Argument 4 to your kernel function is of type Base.Broadcast.Broadcasted{Nothing,Tuple{Base.OneTo{Int64}},typeof(==),Tuple{Base.Broadcast.Extruded{CUDAnative.CuDeviceArray{Int64,1,CUDAnative.AS.Global},Tuple{Bool},Tuple{Int64}},Base.Broadcast.Extruded{Array{Int64,1},Tuple{Bool},Tuple{Int64}}}}.
That type is not isbits, and such arguments are only allowed when they are unused by the kernel.
```

**Pull request merged**